### PR TITLE
Cyclic lr

### PR DESCRIPTION
**Pull request by gandroz**
Just a new scheduler with a modified version of the CyclicLR. It speeds up the training oscillating the LR around the inverse square root used in transducer.
I added a "fix" in the tfrecords generation as TF does not like multiprocessing.

### CODE REVIEW

**Comment by usimarit** [comment]
@gandroz Why don't we use multiprocessing to create tfrecord files? It runs faster. Is there any bug when using it?

**Comment by gandroz** [comment]
> @gandroz Why don't we use multiprocessing to create tfrecord files? It runs faster. Is there any bug when using it?

As I said in my comment, tensorflow crashes when using multiprocessing. I used to have an error similar to this one https://github.com/tensorflow/tensorflow/issues/44137#issuecomment-712824865

**Comment by usimarit** [comment]
@gandroz Perhaps we should do like this [https://github.com/tensorflow/tensorflow/issues/36360](https://github.com/tensorflow/tensorflow/issues/36360)

**Comment by gandroz** [comment]
I could dive deeper in this direction. It could potentially help finding the number of samples per epoch when loading the dataset (which is not possible with tfrecords before the end of the first epoch). I tried to count the number of samples manually, but it takes several long minutes per dataset...

File: tensorflow_asr/optimizers/schedules.py [hunk 1]
```diff
@@ -87,3 +87,36 @@ def __call__(self, step):
...
+
+        self.d_model = d_model
+        self.d_model = tf.cast(self.d_model, tf.float32)
+        self.warmup_steps = warmup_steps
+        self.max_lr = max_lr
+        self.step_size = step_size
+
+    def __call__(self, step):        
+        warmup = step * (self.warmup_steps ** -1.5)
+        lr = 2 * tf.math.rsqrt(step)
```
Comment (usimarit) [inline tensorflow_asr/optimizers/schedules.py:105]: @gandroz You should cast the step to float32 to avoid errors

**Review by usimarit — APPROVED** [review]
